app/src/main/java/net/programmierecke/radiodroid2/players/exoplayer/ExoPlayerWrapper.java

```diff
@@ -102,13 +106,20 @@ public void playRemote(@NonNull OkHttpClient httpClient, @NonNull String streamU
 
         this.context = context;
         this.streamUrl = streamUrl;
+        sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
 
         stateListener.onStateChanged(RadioPlayer.PlayState.PrePlaying);
 
         if (player != null) {
             player.stop();
         }
 
+        backBuffer.clear();
+        int requestedBackBufferSize = 1024 * sharedPref.getInt("backbuffer_size", DEFAULT_BACK_BUFFER_SIZE_KB);
```

Comment:
And another thought - we may know the bitrate of a stream either via shoutcast data or by estimation from ExoPlayer (I didn't look into it). So maybe user facing variable should be in seconds?

Reply:
I was thinking of that, too, but was too lazy so far...